Mk 5056  how to sections (#2)
oduameh/hydra: docs/docs/how-to/commit-blueprint.md

# Commit using a blueprint

> This guide provides a walkthrough on using `cardano-cli` to assemble necessary components for committing funds to a `Head` using a blueprint transaction.

Reviewer: ```suggestion
This guide provides a walkthrough on using `cardano-cli` to assemble the necessary components for committing funds to a `head` using a blueprint transaction.
```

> **Prerequisites**: Access to the hydra-node repository, and the `hydra-node`, `hydra-tui`, `cardano-cli`, and `curl` binaries.

Reviewer: ```suggestion
**Prerequisites**

You should have access to:

- `hydra-node` repository
- `hydra-tui`
- `cardano-cli`, and
- `curl` binaries.

```

## Step 1. Prepare the environment

> Initiate a Cardano-node on the pre production network:

Reviewer: ```suggestion
## Step 1. Prepare the environment
You can use `cardano-cli` to create a _blueprint_ transaction from some `UTXO` you own. First, initiate a Cardano-node on the pre-production network:
```

 ```shell
 ./testnets/cardano-node.sh ~/code/hydra/testnets/preprod
 ```

> ## Step 2. Identify the UTXO
> Determine which `UTXO` you intend to commit to the `Head`. For this example, we use Alice's external wallet key to identify her address:

Reviewer: ```suggestion
## Step 2. Identify the UTXO
Determine which `UTXO` you intend to commit to the `head`. This example uses Alice's external wallet key to identify her address:
```

}
```

> ## Step 3.

Reviewer: Some steps have a title, others don't. This needs to be consistent. I suggest using 'Step 1', 'Step 2', 'Step n'. Note that there shouldn't be a full stop if the step isn't followed by a description.

> Select the first `UTXO`, which has 8 ada available. Use 5 ada to commit and rely on `hydra-node` to balance the commit transaction.

Reviewer: ```suggestion
Select the first `UTXO`, which has eight ada available. Use five ada to commit and rely on `hydra-node` to balance the commit transaction.
```

  --out-file tx.json
```

> ## Step 4.
> Now we have the _blueprint_ transaction in the `tx.json` file. To have `hydra-node` give us a draft commit transaction, we need to:

Reviewer: ```suggestion
## Step 4
You should now have the _blueprint_ transaction in the `tx.json` file. For `hydra-node` to provide a draft commit transaction, you need to:
```

- Obtain the protocol parameters needed to run the `hydra-node`
- Ensure the `hydra-node` is up and running

> - Have the `Head` in the initializing state

Reviewer: ```suggestion
- Have the `head` in the initializing state
```

> - Submit the http request to the `hydra-node` api server using the _blueprint_ transaction we just created and the `UTXO` used for it's input.

Reviewer: ```suggestion
- Submit the HTTP request to the `hydra-node` API server using the _blueprint_ transaction you just created and the `UTXO` used for its input.
```

```

## Step.

> Start the `hydra-node` as a _single_ party Head instance.

Reviewer: ```suggestion
## Step 5
Start the `hydra-node` as a _single_ party head instance.
```

> Note: The value `6264cee4d5eab3fb58ab67f3899ecbcc0d7e72732a2d9c1c5d638115db6ca711` comes from `hydra-node` release [0.16.0](https://github.com/input-output-hk/hydra/releases/tag/0.16.0)

Reviewer: ```suggestion
Note: The value `6264cee4d5eab3fb58ab67f3899ecbcc0d7e72732a2d9c1c5d638115db6ca711` comes from the `hydra-node` release [0.16.0](https://github.com/input-output-hk/hydra/releases/tag/0.16.0).
```

  --node-socket testnets/preprod/node.socket
```

> Press `i` to initialize the `Head`.

Reviewer: ```suggestion
Press `i` to initialize the `head`.
```

> Once the head is in the `Initializing` state, we are ready to send the HTTP request to the `/commit` API path.

Reviewer: ```suggestion
Once the head is in the `Initializing` state, you can send the HTTP request to the `/commit` API path.
```

> Assemble the request body using the `cborHex` field from the tx-body file `tx.json` and the JSON representation of the `UTXO` we used as input.

Reviewer: ```suggestion
Assemble the request body using the `cborHex` field from the tx-body file `tx.json` and the JSON representation of the `UTXO` you used as input.
```

}
```

Save this JSON to a `commit-request.json` file.

> Now, please prompt the running `hydra-node` to draft a commit transaction for us:

Reviewer: ```suggestion
You can now prompt the running `hydra-node` to draft a commit transaction:
```

```

> This yields a large CBOR blob which we can save to `commit-tx.json` file.

Reviewer: ```suggestion
This yields a large CBOR blob, which you can save to the `commit-tx.json` file.
```

> Now we need to sign and submit the draft commit transaction.

Reviewer: ```suggestion
Next, sign and submit the draft of the commit transaction:
```

Reviewer: Change the last sentence below to:

'If you start the `hydra-tui` and wait until the transaction you just sent is re-observed by the `hydra-node`, you should see that the head is now open.
